javascript/javascript2/week2/hyfBayHelper/main.js

```diff
@@ -0,0 +1,61 @@
+
+  console.log("Script loaded");
+
+const products = getAvailableProducts();
+console.log(products);
+
+
+
+const productsList = document.getElementById('productsList');
+ const search= document.getElementById('search-item');
+
+const searchPrice= document.getElementById('searchPrice')
+
+
+function renderProducts(products) {
+
+const productsList = document.querySelector('.productsList');
+  
+ products.forEach(product => {
+  
+   const  li= document.createElement('li');
+      li.innerHTML = `
+   
+     <li><strong> ${product.name} </strong> </li>
+  <li> price : ${product.price} </li>
+  <li>  Rating : ${product.rating} </li>
+
+  ` ;
+
+   productsList.appendChild(li);
+  console.log(product); 
+
+});
+
+ }; 
+
+ renderProducts(products);
+
+  search.addEventListener ('input', (item )=>{
+  //console.log(item.target.value); 
+  let value =item.target.value.toLowerCase();
+ const filteredProducts= products.filter((product)=>{
+   return product.name.toLowerCase().includes(value); 
+});
+  console.log(filteredProducts); 
+  renderProducts(filteredProducts);
+ 
+}); 
+
+
+ searchPrice.addEventListener('input' , (e)=> {
```

Comment:
This function should looks like this `searchPrice.addEventListener("input", (e) => {
  const currentPrice = e.target.value;
  const filteredPrice = products.filter(
    (product) => product.price <= currentPrice
  );
  filteredPrice.sort((a, b) => a.price - b.price);

  renderProducts(filteredPrice);
});`